src/components/popover/bl-popover.ts

```diff
@@ -0,0 +1,200 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import {
+  computePosition,
+  flip,
+  shift,
+  offset,
+  arrow,
+  inline,
+  autoUpdate,
+  size,
+  Middleware, MiddlewareState,
+} from '@floating-ui/dom';
+import { classMap } from 'lit/directives/class-map.js';
+import style from './bl-popover.css';
+import { event, EventDispatcher } from '../../utilities/event';
+
+export type Placement =
+  | 'top-start'
+  | 'top'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom'
+  | 'bottom-end'
+  | 'left-start'
+  | 'left'
+  | 'left-end'
+  | 'right-start'
+  | 'right'
+  | 'right-end';
+
+/**
+ * @tag bl-popover
+ * @summary Baklava Popover component
+ *
+ * @property {string} placement - Sets the popover placement
+ */
+@customElement('bl-popover')
+export default class BlPopover extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @query('.popover') private popover: HTMLElement;
+  @query('.arrow') private arrow: HTMLElement;
+
+  /**
+   * Sets placement of the popover
+   */
+  @property({ type: String })
+  placement: Placement = 'top';
+
+  @state() _target: string | Element;
+
+  @property({ type: Boolean })
+  shift = false;
+
+  @property({ type: Boolean, attribute: 'fit-size' })
+  fitSize = false;
+
+  @property({ type: Number })
+  offset = 10;
+
+  @state() private _visible = false;
+
+  /**
+   * Fires when hovering over a trigger
+   */
+  @event('bl-popover-show') private onBlPopoverShow: EventDispatcher<string>;
+
+  /**
+   * Fires when leaving over from trigger
+   */
+  @event('bl-popover-hide') private onBlPopoverHide: EventDispatcher<string>;
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+
+    this.popoverAutoUpdateCleanup && this.popoverAutoUpdateCleanup();
+  }
+  private _middleware: Middleware[] = [];
+
+  protected firstUpdated() {
+    this._middleware.push(
+      offset(this.offset),
+      flip(),
+      inline(),
+      arrow({ element: this.arrow, padding: 5 })
+    );
+    if (this.shift) this._middleware.push(shift({ padding: 5 }));
+
+    if (this.fitSize) {
+      this._middleware.push(
+        size({
+          apply(args: MiddlewareState) {
+            if(args.elements.floating && args.elements.reference){
+              Object.assign(args.elements.floating.style, {
+                width: `${args.elements.reference.getBoundingClientRect().width}px`,
+              });
+            }
+          },
+        })
+      );
+    }
```

Comment:
Because of those parameters (`shift`, `fitSize`) can be changed during time, setting them in `firstUpdated` will not cover all use cases. I would move all of those middleware definitions to a method that returns the list of middlewares according to current parameters, and would call it inside setPopover method.